app/jobs/cleanup_account_job.rb

```diff
@@ -1,18 +1,27 @@
 class CleanupAccountJob < ActiveJob::Base
   def perform(account)
-    cleanup_solr(account)
-    cleanup_fedora(account)
+    account.switch do
+      cleanup_solr(account)
```

Comment:
solr and fedora cleanup should probably stay *outside* the switch so that the mechanism by which they operate is clear.  This also emphasizes the point of your comment that the switch is for redis.